### PR TITLE
Add ACPI downloader

### PR DESCRIPTION
**Pull request by AlexandreAbraham**
This is only a draft. No Guarantee of functioning for all pieplines, missing doc.

### CODE REVIEW

File: nilearn/datasets/func.py [hunk 1]
```diff
@@ -1557,3 +1557,120 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
...
+
+    Parameters
+    ----------
+    n_subjects: int, optional
+        Number of subjects to load. Maximum is 129. If n_subjects is None, all
+        available subjects are loaded.
+
+    preprocessing: string, optional
+        Preprocessing pipeline to use. Possible values are "ANTS" and "FNIRT".
+        ANTS is the default.
```
Comment (aabadie) [inline nilearn/datasets/func.py]: Minor. I would use the same phrasing as below: `Default: "ANTS"`.

File: nilearn/datasets/func.py [hunk 2]
```diff
@@ -1557,3 +1557,120 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
...
+
+    scrubbing: boolean, optional
+        Whether or not to load scrubbed dataset. Default is False.
```
Comment (aabadie) [inline nilearn/datasets/func.py]: Same comment (still minor)

File: nilearn/datasets/func.py [hunk 3]
```diff
@@ -1557,3 +1557,120 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
...
+                         verbose=verbose)[0]
+
+    html = open(files, "r").read()
+
+    # Prepare the regexp to get id ranges
+    root = preprocessing.lower() + '_s%i_g%i' % (
+        scrubbing, global_signal_regression)
+    regexp = root + '_([0-9]+)_([0-9]+)'
+    indices = re.findall(regexp, html)
+    indices = [(int(s), int(e)) for (s, e) in indices]
```
Comment (aabadie) [inline nilearn/datasets/func.py]: I guess `s` is for subject and `e` is for experiment. I like explicit variable names ;)

File: nilearn/datasets/func.py [hunk 4]
```diff
@@ -1557,3 +1557,120 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
...
+
+    csv = csv[:n_subjects]
+    ids = csv['SUBID']
+
+    # Now, download the files
+    opts = {'uncompress': True}
+    func = []
+    for i in ids:
+        # Find the file where the subject data is located
+        s, e = [(s, e) for (s, e) in indices if s <= i and i <= e][0]
```
Comment (aabadie) [inline nilearn/datasets/func.py]: Same comment as above here.

**Comment by AlexandreAbraham** [comment]
Problem : some archives do not contain all subject data. Plus, some subject data is available without being listed in the phenotypic file. How should we handle this?

**Comment by AlexandreAbraham** [comment]
This is a bit hacky but this should do the job. Can you tell me if it works for the whole dataset?

**Comment by aabadie** [comment]
I gave a try to this change, like this
```python
from nilearn.datasets import fetch_acpi
dataset = fetch_acpi()
```

It seems to work. A few comments:
* `fetch_acpi` should be imported and listed in the `__all__` variable of the `__init__.py`of the datasets module.
* I had a warning at the very beginning of the fetch: 
`Warning: Could not find dataset description.`

Download is still in progress. Will give more feedback later.

**Comment by AlexandreAbraham** [comment]
What you posted is the same I tested and should work. But I don't have enough disk space to test

```python
from nilearn.datasets import fetch_acpi
dataset = fetch_acpi(n_subjects=None)
```

I asked Kamalakar because he has already downloaded everything.

Regarding the other issues, this is normal. This PR is a quickie to allow Kamalakar to download the dataset. Proper documentation will come later.

**Comment by aabadie** [comment]
> What you posted is the same I tested and should work.

My try failed with an "OSError File already exists" (or something similar). Sorry, I don' t have the full message output (lost it...).

**Comment by KamalakerDadi** [comment]
Should Fix #782

**Comment by KamalakerDadi** [comment]
>My try failed with an "OSError File already exists" (or something similar).

Since data extraction always find the same folder for each bunch of 10 subjects download. For example, I tried to fetch FNIRT_S0_G0. 
```python
IOError: Fetching aborted: [Errno 17] File exists: '/volatile/home/kamalakar/work/acpi/77d20ae3d4678e62721b461c115dac94/fnirt_s0_g0'
```

**Comment by AlexandreAbraham** [comment]
The problem of `File exists` should be solved now.

File: nilearn/datasets/func.py [hunk 5]
```diff
@@ -1557,3 +1558,141 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
...
+        - 'func': string list
+            Paths to Nifti images.
+        - 'phenotypic': ndarray
+            Contains phenotypic data.
+        - 'description': data description of the release and references.
+
+    Notes
+    -----
+    More information about datasets structure, See:
+    https://figshare.com/articles/COBRE_preprocessed_with_NIAK_0_12_4/1160600
```
Comment (KamalakerDadi) [inline nilearn/datasets/func.py]: Notes section is misleading to COBRE.

**Comment by kchawla-pi** [comment]
Is this alive? If not, please close it by this weekend.

**Comment by KamalakerDadi** [comment]
Hi @AlexandreAbraham Do you think we should still keep this work on the fetcher or close ?

I am afraid of maintenance on the data fetchers. See #1824 .
Also, I don't see that it can be replaced with light weight dataset we have on the examples.

**Comment by GaelVaroquaux** [comment]
+1 for closing.

**Comment by AlexandreAbraham** [comment]
Kill it with fiiiirrrreeee

**Comment by KamalakerDadi** [comment]
:fire:

**Comment by GaelVaroquaux** [comment]
> 🔥

:)

**Pull request closed without merging**